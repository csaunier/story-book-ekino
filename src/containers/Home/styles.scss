.home {
  align-items: center;
  display: flex;
  flex: 1;
  justify-content: center;
}

.home__logo {
  align-items: center;
  border-radius: 50%;
  box-shadow: 0 0 10px 2px #bbb;
  display: flex;
  height: 200px;
  justify-content: center;
  width: 200px;
}
