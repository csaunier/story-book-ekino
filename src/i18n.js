import { reactI18nextModule } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import XHR from 'i18next-xhr-backend'

const { FALLBACK_LOCALE } = process.env

const loadLocales = (url, options, callback) =>
  require(`bundle-loader!i18n/${url}.yml`)(locale => callback(locale, { status: '200' }))

i18n
  .use(XHR)
  .use(LanguageDetector)
  .use(reactI18nextModule)
  .init({
    backend: {
      loadPath: '{{lng}}',
      parse: data => data,
      ajax: loadLocales,
    },

    detection: {
      order: ['querystring', 'htmlTag'],
    },

    fallbackLng: FALLBACK_LOCALE,

    interpolation: {
      escapeValue: false,
    },

    react: {
      wait: true,
    },

    debug: !!__DEBUG__,
  })

export default i18n
