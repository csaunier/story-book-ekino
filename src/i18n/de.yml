menu:
  home: Startseite
  example: Beispiel

example:
  content: >
    Hey! Ich bin eine Beispielseite

switchLanguage:
  fr: Franz√∂sisch
  de: Deutsch
  en: Englisch
