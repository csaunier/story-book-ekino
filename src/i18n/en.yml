menu:
  home: Home
  example: Example

example:
  content: >
    Hey! I am an example page

switchLanguage:
  fr: French
  de: German
  en: English
