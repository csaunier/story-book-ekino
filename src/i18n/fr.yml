menu:
  home: Accueil
  example: Exemple

example:
  content: >
    Hey ! Je suis une page d'exemple

switchLanguage:
  fr: Fran√ßais
  de: Allemand
  en: Anglais
