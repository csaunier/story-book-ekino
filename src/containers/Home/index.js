import React, { PureComponent } from 'react'

import random from 'lodash/random'
import { Motion, spring } from 'react-motion'
import { Helmet } from 'react-helmet'
import { translate } from 'react-i18next'

import ekinoLogo from 'assets/img/ekino.png'

import styles from './styles.scss'

const mySpring = value => spring(value, { stiffness: 100, damping: 50 })
const getRgb = ({ r, g, b }) => `rgb(${parseInt(r, 10)}, ${parseInt(g, 10)}, ${parseInt(b, 10)})`

class Home extends PureComponent {
  state = {
    r: 0,
    g: 0,
    b: 0,
    x: 0,
    y: 0,
  }

  componentDidMount() {
    this.setAnimationValues()
  }

  componentWillUnmount() {
    clearTimeout(this._timeout)
  }

  setAnimationValues(delay = 0) {
    this._timeout = setTimeout(
      () =>
        window.requestAnimationFrame(() => {
          this.setState({
            r: random(0, 255),
            g: random(0, 255),
            b: random(0, 255),
            x: random(-10, 10),
            y: random(-10, 10),
          })

          this.setAnimationValues(random(100, 500))
        }),
      delay,
    )
  }

  _timeout = null

  render() {
    const { t } = this.props
    const { r, g, b, x, y } = this.state

    return [
      <Helmet key="helmet">
        <title>{t('menu.home')}</title>
      </Helmet>,
      <div key="content" className={styles.home}>
        <Motion
          style={{
            r: mySpring(r),
            g: mySpring(g),
            b: mySpring(b),
            x: mySpring(x),
            y: mySpring(y),
          }}
        >
          {m => (
            <div
              className={styles.home__logo}
              style={{
                backgroundColor: getRgb({ r: m.r, g: m.g, b: m.b }),
                transform: `translate3d(${m.x}px, ${m.y}px, 0)`,
              }}
            >
              <img src={ekinoLogo} alt="Ekino" />
            </div>
          )}
        </Motion>
      </div>,
    ]
  }
}

export default translate()(Home)
