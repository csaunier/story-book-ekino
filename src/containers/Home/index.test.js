import React from 'react'

import { shallow } from 'enzyme'
import { I18nextProvider } from 'react-i18next'

import i18n from 'i18n'

import Home from './'

describe('component Home', () =>
  it('should render', () => {
    const component = shallow(
      <I18nextProvider i18n={i18n}>
        <Home />
      </I18nextProvider>,
    )

    expect(component).toMatchSnapshot()
  }))
