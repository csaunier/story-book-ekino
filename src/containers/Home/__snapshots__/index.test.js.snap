// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Home should render 1`] = `<Translate(Home) />`;
